devtools/common/src/main/java/io/quarkus/generators/SourceType.java

```diff
@@ -68,4 +73,8 @@ private String computeTemplateFile(String genericTemplate, String templateName)
     private String computeTemplateFile(String genericTemplate, String templateName, String fileName) {
         return String.format(genericTemplate, templateName, getPathDiscriminator(), fileName);
     }
+
+    public static Optional<SourceType> parse(String extension) {
+        return Arrays.stream(values()).filter(e -> extension.contains(e.name().toLowerCase())).findAny();
```

Comment:
I really would prefer us having a match method for each `SourceType`. Using the enum name is really fragile as we could change it without thinking it would break the behavior.

Reply:
I have add an another field `name` in `SourceType` to maintain name of the extension.